### ISSUE TITLE
error: could not perform API request: http error: [55] Failed sending data to the peer

### ISSUE THREAD

**Issue by xispower** [getsentry/sentry-cli#7]
The last question was not resolved,
Client log:
[INFO] sentrycli::api using token authentication
[INFO] sentrycli::api sending form data
error: could not perform API request: http error: [55] Failed sending data to the peer

Server log:
Invalid (too big) CONTENT_LENGTH. skip
"POST /api/0/projects/sentry/test/files/dsyms/ HTTP/1.1" 500 0 "-" "sentry-cli/0.18.0"

**Comment by mitsuhiko** [comment]
Is this still happening?

**Comment by mitsuhiko** [comment]
Going to close this on inactivity.

**Comment by ahti** [comment]
For reference of ppl stumbling upon this issue while googling: This is expected behaviour, since the upload exceeds uwsgis maximum post size ([20mb by default](https://github.com/getsentry/sentry/blob/461204944f5a6c13159b1a4ff0b6849c17d1daa7/src/sentry/services/http.py#L66)). You can change your sentry.conf.py to include `'limit-post': <number of bytes>` in `SENTRY_WEB_OPTIONS` to adjust the maximum post size.

**Comment by mcrawshaw** [comment]
@ahti Does this assume you are running your own server?

**Comment by ahti** [comment]
@mcrawshaw Yes. I don't know how or whether this would occur/could be fixed on hosted Sentry.

**Comment by igorpejic** [comment]
This is what worked for me with a docker-compose setup:
change the `Dockerfile` to the newest sentry image (at my time it was 8.22-onbuild):

```
FROM sentry:8.22-onbuild
```

In `sentry.conf.py` change the `SENTRY_WEB_OPTIONS` to:

```
SENTRY_WEB_OPTIONS = {
    # 'workers': 3,  # the number of web workers
    'lmit-post': 31457280  # in bytes ~ 30MB
}
```

Stop the containers: `docker-compose stop`
Run the upgrade: `docker-compose run --rm web upgrade`
Start the containers again: `docker-compose up -d`

If you are serving sentry behind a `nginx proxy`, you also need to add:
`client_max_body_size 30m;` 
to your `nginx.conf/sites-enabled` where your site is served.

**Comment by jan-auer** [comment]
FWIW, the next sentry release (9.0.0) will contain a permanent server-side fix for this.

**Comment by pengshaosu** [comment]
@jan-auer 
Has the issue fixed in version 9.0.0?

I has the self-hosted sentry(9.0.0), but found that also need set the sentry.conf.py to resolve the issue.
```
SENTRY_WEB_OPTIONS = {
    # 'workers': 3,  # the number of web workers
    'lmit-post': 31457280  # in bytes ~ 30MB
}
```
thx
